### PR TITLE
validate pool owner witness

### PR DESCRIPTION
**Pull request by mkv-vcm**
Not sure how else we should test this feature, I welcome recommendations.

### CODE REVIEW

File: test/integration/__fixtures__/signTxPoolRegistration.ts [hunk 1]
```diff
@@ -692,6 +693,27 @@ export const poolRegistrationOwnerTestcases: Testcase[] = [
...
+    testname: string,
+    tx: Transaction,
+    additionalWitnessPaths: BIP32Path[],
+    errCls: new (...args: any[]) => ErrorBase,
+    errMsg: string,
+}
+
+export const poolRegistrationOwnerRejectTestcases: RejectTestcase[] = [
+    {
+        testname: "Witness valid multiple mixed owners all relays pool registration with additional witnesses",
```
Comment (janmazak) [inline test/integration/__fixtures__/signTxPoolRegistration.ts:706]: The name seems wrong. Also, no need to mention details about owners being mixed or anything about relays.

File: test/integration/__fixtures__/signTxPoolRegistration.ts [hunk 2]
```diff
@@ -692,6 +693,27 @@ export const poolRegistrationOwnerTestcases: Testcase[] = [
...
+        tx: {
+            ...txBase,
+            certificates: [certificates.poolRegistrationMixedOwnersAllRelays],
+        },
+        additionalWitnessPaths: [str_to_path("1852'/1815'/0'/2/0"), str_to_path("1852'/1815'/0'/2/1"), str_to_path("1854'/1815'/0'/2/0")],
```
Comment (janmazak) [inline test/integration/__fixtures__/signTxPoolRegistration.ts:711]: I'd split this into two tests, one with inconsistent 1852 path, another with a 1854 path (with the only discrepancy being 1852 for the owner, 1854 for the witness, the rest of path equal).

File: test/integration/signTxPoolRegistration.test.ts [hunk 3]
```diff
@@ -47,6 +48,30 @@ describe("signTxPoolRegistrationOKOwner", async () => {
     test(poolRegistrationOwnerTestcases, TransactionSigningMode.POOL_REGISTRATION_AS_OWNER)
 })
 
+describe("signTxPoolRegistrationNotOKOwner", async () => {
```
Comment (janmazak) [inline test/integration/signTxPoolRegistration.test.ts:51]: Maybe something about rejecting tx with incorrect pool registration owner witness.

File: test/integration/__fixtures__/signTxPoolRegistration.ts (hunk 1 above)
Comment (mkv-vcm) [reply test/integration/__fixtures__/signTxPoolRegistration.ts:706]: The name comes from another (positive) test (Witness valid multiple mixed owners all relays pool registration) that was modified with additional witnesses. That test's name comes from the used certificate poolRegistrationMixedOwnersAllRelays.

File: test/integration/signTxPoolRegistration.test.ts (hunk 3 above)
Comment (mkv-vcm) [reply test/integration/signTxPoolRegistration.test.ts:51]: Will have to be inserted into the test case rejection PR when that one gets rebased on top of this anyway.

**Review by janmazak — APPROVED** [review]
Apart from lint issues, it looks OK.

**Pull request merged**